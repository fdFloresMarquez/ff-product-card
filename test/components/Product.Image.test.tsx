import React from 'react';
import renderer from 'react-test-renderer';

import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('Product Image', () => {
    test('should render without crashing with a custom image', () => {
      const wrapper = renderer.create(
        <ProductCard product={product2}>{() => <ProductImage img='https://hola.com' />}</ProductCard>
      );
  
      expect( wrapper.toJSON() ).toMatchSnapshot();
    });
    
    test('should render without crashing productImage', () => {
      const wrapper = renderer.create(
        <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
      );
  
      expect( wrapper.toJSON() ).toMatchSnapshot();
    });
  });