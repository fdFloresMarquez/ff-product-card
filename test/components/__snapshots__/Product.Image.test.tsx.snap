// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image should render without crashing productImage 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="CoffeeMug"
  />
</div>
`;

exports[`Product Image should render without crashing with a custom image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="https://hola.com"
  />
</div>
`;
